### ISSUE TITLE
[Feature Request] Return a list of events based at particular point in time

### ISSUE THREAD

**Issue by ashkuren** [niccokunzmann/python-recurring-ical-events#11]
Hello,

It would be nice to have a function that takes a datetime object as input and returns a list of events that are active at specific moment.

As a use case, we have a calendar with blackouts (represented as events). On incoming alert from monitoring system, we should check if there is a blackout on specific server at the moment. If blackout found, log actions and drop alert.

This can be something like this:
```python
    def at_moment(self, now: datetime.datetime = None) -> List[Event]:
        """This function should return the list of events active at the specific point in time"""
        if not now:
            now = datetime.datetime.now()
        # If event starts somewhere previous day with long duration
        start = now - datetime.timedelta(days=1)
        end = now + datetime.timedelta(days=1)

        events = self.between(start.date().timetuple()[:3], end.date().timetuple()[:3])
        return [e for e in events if e.get('dtstart').dt < now < e.get('dtend').dt]
```
I'm not sure what's better way to identify the start and the end for `between` function. If we'll have an event with duration more than one day, it will not work.

**Comment by niccokunzmann** [comment]
Hi and welcome.
I agree this is a good idea. Part of this is implemented. Did you check that overlapping events are not returned?
I think, between(now, now) would return the events which occur now.

Looking at the tests. This file is empty: https://github.com/niccokunzmann/python-recurring-ical-events/blob/master/test/test_span_overlap.py and can be deleted. This file tests the overlap function: https://github.com/niccokunzmann/python-recurring-ical-events/blob/master/test/test_span_is_in_duration.py
Also, including events overlapping start and stop is the default used:
https://github.com/niccokunzmann/python-recurring-ical-events/blob/43a00da8337896101ff9f02b1bb4e6cc93b93f2f/recurring_ical_events.py#L44
https://github.com/niccokunzmann/python-recurring-ical-events/blob/43a00da8337896101ff9f02b1bb4e6cc93b93f2f/recurring_ical_events.py#L111

`at(now)` should do it, then.

Since you opened an issue, I think, this is really not clear and so I value your contribution also in form of documentation.
- What do you think can be changed to make this more clear?
- Is there a test case which does not fulfill your requirements?

**Comment by ashkuren** [comment]
Hey,

Thanks for your response. I have done some tests on my side and `between` function works fine with double `datetime` object as input.

For me, the function `between` means that it will return all the events that fall into that defined period/span (as it's stated in docstring). It does not explicitly states, that it can also work in reverse: give me all events that occur at specific `datetime`. And moreover, oddly you need to put same `datetime` twice for `start` and `stop` parameters.

For me, it sould be more explicit. For example putting an alias function like this:
```python
    def at_dt(self, dt: datetime):
        return self.between(dt, dt)
```
Or maybe, allowing `at` function to accept `datetime` as input. As of now, following test fails:
```python
def test_at_for_datetime(calendars):
    now = datetime(2018, 1, 15, 10, 30, 0, 0)
    events = calendars.duration.at(now)
    assert len(events) == 1
    assert events[0].get('summary') == 'Duration Event'
```
Pytest output:
```
self = <recurring_ical_events.UnfoldableCalendar object at 0x000001A8DA6AC748>, date = datetime.datetime(2018, 1, 15, 10, 30)

    def at(self, date):
        """Return all events within the next 24 hours of starting at the given day.

        - date can be a year like (2019,) or 2099
        - a month like (2019, 1) for January of 2019
        - a day like (2019, 1, 1) for the first of January 2019
        - a day with hours, (2019, 1, 1, 1)
        - a day with minutes, (2019, 1, 1, 1, 1)
        - a day with seconds, (2019, 1, 1, 1, 1, 1)
        """
        if isinstance(date, int):
            date = (date,)
        if isinstance(date, str):
            assert len(date) == 8 and date.isdigit(), "format yyyymmdd expected"
            date = (int(date[:4], 10), int(date[4:6], 10), int(date[6:]))
>       if len(date) == 1:
E       TypeError: object of type 'datetime.datetime' has no len()

recurring_ical_events.py:265: TypeError
```

**Comment by niccokunzmann** [comment]
Hi,

uh. at(dt) should work! Yes. Can you create a pull request with this
test, so it fails?
Someone can fix it, then.

If you like, feel free to edit the description and make it more clear.
My understanding of between was this:
If I am a week on holiday, I can ask: Are you on holiday bewteen Monday
and Thuesday?
Yes - the event is included.

What do you this needs to be done so, it fits your interest?

Best,
Nicco

**Comment by niccokunzmann** [comment]
This fixed from version v0.1.11b0 on.
